cpp_tests/support/thread_local_test.cc

```diff
@@ -30,8 +30,17 @@ namespace {
 // TheadLocal storage can properly handle such types.
 struct WorkerId {
   explicit WorkerId(int id) : id(id) {}
+
+  // Workaround for an apparent MSVC bug which results in a build failure
+  // if WorkerId is move-only
+#ifdef _MSC_VER
```

Comment:
I have spent quite a bit trying to figure out what's going on, but it proved too time consuming. The gist is that the combination of MSVC compiler, standard library (`std::vector`) and `llvm::Optional` wrapping a move-only type doesn't build. For some reason MSVC compiler insists on instantiating the copy constructor for `llvm::Optional` but I don't see anything in the code to explain it (and clang/gcc seem to be like the code just fine)

My best idea to move forward is the `#ifdef`, but I'm open to better ideas.

Reply:
The test specifically checks that movable-only types are supported. I'm wondering if we just hit the same issue elsewhere if we skip that part of the test for MSVC.

But admittedly, I'm also quite confused. I can reproduce this on clang: https://godbolt.org/z/PzsWvGE85. Not sure why and what the difference to the code here is.

Reply:
Maybe we can just use llvm::SmallVector instead of std::vector? This works for clang in compiler explorer (but I failed to try it with msvc, I don't know how to include llvm headers).

Reply:
Thanks - using `llvm::SmallVector` is an interesting idea, although I'm pretty sure `std::vector` handles move-only types just fine. Unless it's a MSVC compiler bug, the problem may be with `llvm::Optional` and replacing `llvm::Optional` with `std::optional` may be another option. Does the current Google C++ style & TFRT accept C++17?